src/hash.rs

```diff
@@ -64,3 +70,65 @@ fn file_hash(algorithm: &str, path: &str) -> Result<String> {
 
     Ok(hash_algorithm(algorithm, &bytes)?)
 }
+
+fn totp_generate(hex_seed: &str) -> Result<String> {
+    let mut seed: [u8; 64] = [0; 64];
+
+    let hex_seed_bytes = (0..hex_seed.len())
+                        .step_by(2)
+                        .map(|i| u8::from_str_radix(&hex_seed[i..i + 2], 16));
+
+    let mut i = 0;
+    for b in hex_seed_bytes {
+        seed[i] = b.unwrap();
+        i += 1;
+    }
+
+    let mut ipad: [u8; 64] = [0; 64];
+    let mut opad: [u8; 64] = [0; 64];
+
+    for j in 0..64 {
+        ipad[j] = seed[j] ^ 0x36;
+        opad[j] = seed[j] ^ 0x5C;
+    }
```

Comment:
Since Rust 1.55.0 (which our CI is using) you can replace the loop and declarations here with:
```rust
let mut ipad = seed.map(|x| x ^ 0x36);
let mut opad = seed.map(|x| x ^ 0x5C);
```

I'm also not sure these need to be mutable.